apis/config/v1beta2/types.go

```diff
@@ -148,6 +148,8 @@ type NodeResourceTopologyMatchArgs struct {
 	// implicitely enables the caching. If zero, disables the caching entirely.
 	// If the cache is enabled, the Reserve plugin must be enabled.
 	CacheResyncPeriodSeconds *int64 `json:"cacheResyncPeriodSeconds,omitempty"`
+	// if set to true, exclude node from scheduling if there are any reserved pods for given node
+	DiscardReservedNodes bool `json:"discardReservedNodes,omitempty"`
```

Comment:
my only remaining concern is this effectively makes the code ignore `CacheResyncPeriodSeconds` (if given).
I'd start documenting this. We can think about a better UX later on.